src/sql/parts/modelComponents/componentBase.ts

```diff
@@ -46,7 +46,9 @@ export abstract class ComponentBase extends Disposable implements IComponent, On
 	protected _onEventEmitter = new Emitter<IComponentEventArgs>();
 
 	public layout(): void {
-		this._changeRef.detectChanges();
+		if (!this._changeRef['destroyed']) {
```

Comment:
here I'm trying to see if the component is being disposed and not to refresh it. this seems to work but if there's a better way to do it please let me know.